x-pack/plugins/files/public/components/file_picker/file_picker_state.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  map,
+  tap,
+  from,
+  switchMap,
+  Observable,
+  shareReplay,
+  debounceTime,
+  Subscription,
+  combineLatest,
+  BehaviorSubject,
+  distinctUntilChanged,
+} from 'rxjs';
+import { FileJSON } from '../../../common';
+import { FilesClient } from '../../types';
+
+function naivelyFuzzify(query: string): string {
+  return query.includes('*') ? query : `*${query}*`;
```

Comment:
Another approach could be that we are more strict, and don't expose Elasticsearch specific things, like `*` operator. Instead, we strip all `*` operators from user's input, and on the server we can wrap the search text in the `*` operators.